controllers/users.js

```diff
@@ -466,6 +468,74 @@ const filterUsers = async (req, res) => {
   }
 };
 
+const DISCORD_BASE_URL = config.get("services.discordBot.baseUrl");
+
+// wroker queue created with Bull
+const syncQueue = new Queue("syncQueue", {
+  redis: {
+    port: REDIS_PORT,
+    host: REDIS_URI,
+  },
+});
+
+/*
+ * Sync user data with the data present in our discord server
+ *
+ */
+
+const syncInDiscordRole = async (req, res) => {
+  try {
+    // create a authToken to verify on discord bot
+    const authToken = jwt.sign({}, config.get("rdsServerlessBot.rdsServerLessPrivateKey"), {
+      algorithm: "RS256",
+      expiresIn: config.get("rdsServerlessBot.ttl"),
+    });
+
+    // fetch the data of member in discord server
+    const response = await fetch(`${DISCORD_BASE_URL}/discord-members`, {
+      headers: {
+        "Content-Type": "application/json",
+        Authorization: `Bearer ${authToken}`,
+      },
+    });
+    const discordMembers = await response.json();
+
+    // get the data of the users from our database
+    const allUsers = await userQuery.getAllUsers();
+
+    allUsers.forEach((doc) => {
+      const user = { id: doc.id, ...doc.data() };
+
+      if (user.roles.archived) {
+        // Update: If the user is archived change the role in_discord: false
+        user.roles = { ...user.roles, in_discord: false };
+        syncQueue.add({ user });
```

Comment:
shall we check before updating if the `in_discord ` is true.

Reply:
done